src/autoval/lib/host/bmc.py

```diff
@@ -84,9 +84,21 @@ def get_fru_name(self) -> str:
         return fru
 
     def _get_slot(self) -> str:
+        """
+        Retrieve the slot number of the host.
+
+        Returns:
+            The slot number of the host.
+        Raises:
+            TestInputError: if the slot cannot be found
+        """
         slot_details = self.run(f"/usr/local/bin/slot-util {self.host.hostname}")
-        slot_number = slot_details.splitlines()[1].strip().split(":")[0]
-        return slot_number
+        if re.search(r"\bslot\d+\b", slot_details):
+            slot_number = slot_details.splitlines()[1].strip().split(":")[0]
```

Comment:
Consider simplifying this

```diff
-         slot_details = self.run(f"/usr/local/bin/slot-util {self.host.hostname}")
-         if re.search(r"\bslot\d+\b", slot_details):
-             slot_number = slot_details.splitlines()[1].strip().split(":")[0]
+         slot_details = self.run(f"/usr/local/bin/slot-util all")
+         match = re.search(rf"(slot\d+): {self.host.hostname}", slot_details)
+         if match:
+             slot_number = match.group(1)
```

This will make it easier to debug because the `slot-util all` output (which contains the complete `slot` --> `hostname` mapping) will be available in the command log.